Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -104,11 +104,15 @@ internal sealed class RestChannel : ChannelBase
             Func<Status> statusFunc = () => GetStatus(statusResponseSource.Task);
             Func<Metadata> trailersFunc = () => GetTrailers(statusResponseSource.Task);
 
+            // The response stream will dispose of the cancellation context if:
+            // - The user disposes of it directly
+            // - Any exception is thrown while reading an element
+            // - We reach the last element
             PartialDecodingStreamReader<TResponse> responseStream = new PartialDecodingStreamReader<TResponse>(readerTask, restMethod.ParseJson<TResponse>, cancellationContext);
             Action disposalAction = () =>
             {
-                responseStream.Dispose();
                 cancellationContext.Cancel();
```

Comment:
That's the intention of disposing of the call. From the gRPC docs:

> If the call has already finished normally (response stream has been fully read), doesn't do anything. Otherwise, requests cancellation of the call which should terminate all pending async operations associated with the call.